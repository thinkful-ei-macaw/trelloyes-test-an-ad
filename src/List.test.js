import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List.js';

// Smoke Test
describe('<List />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List
            header="Some header"
            cards={[
            {id:"1", title: "Card One", content: "lorem ipsum"},
            {id:"2", title: "Card Two", content: "Some other lorem ipsum"},
            {id:"3", title: "Card Three", content: "Some different lorem ipsum"}
            ]}
        />, div);
        ReactDOM.unmountComponentAtNode(div);
    })
})

// Snapshot Test
describe('<List />', () =>{
    it('render this UI as expected', () => {
        const tree = renderer.create(<List
            header="Some header"
            cards={[
              {id:"1", title: "Card One", content: "lorem ipsum"},
              {id:"2", title: "Card Two", content: "Some other lorem ipsum"},
              {id:"3", title: "Card Three", content: "Some different lorem ipsum"}
              ]}
          />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

// header -> string
// cards -> array of objects with id, title and content (all strings)

