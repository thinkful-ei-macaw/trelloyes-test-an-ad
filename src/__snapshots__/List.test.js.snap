// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> render this UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Some header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Card One
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Card Two
      </h3>
      <p>
        Some other lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Card Three
      </h3>
      <p>
        Some different lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
